### ISSUE TITLE
How do I add a project programmatically?

### ISSUE THREAD

**Issue by deitch** [Codiad/Codiad#877]
How do I add a project programmatically from outside of Codiad? Is there a simple way to 

1. Create a project
2. Add it to restricted users?

**Comment by cheiff** [comment]
The idea is:
- Users
  - Read the users.php (json file) into an variable (use the getJSON function) 
    ```php
$users = getJSON('users.php');
```
  -  Add your new user
```php
$user = array(
  'username' => 'john',
  'password' => sha1(md5('supersecretpassword')),
  'project' => 'myproject'
);
$users[] = $user;
```
  - Save the file 
```php
saveJSON('users.php',$users);
```
- Project
  - Read the projects.php (json file) into an array (use the getJSON function) 
  -  Add your new project
  - Save the file
- ACL
  - Create a new username_acl.php file
  - encode in json an array with the workspaces

**Comment by deitch** [comment]
I think I get it. So the ACL file lists which users have which rights over which projects? If so, what is the `'project' => 'myproject'` in the user file you showed above?

**Comment by deitch** [comment]
OK, I tried it in the UI and then opened the files. I see the users, projects, and *username*_acl files. The one I set up as restricted has an *username*_acl file listing the only projects to which I gave him access. The rest do not. That makes sense.

The projects file lists all the projects, so that makes sense too.

But there is still one project listed in each person's hash in users.php. What is that?
